API/hermes/CMakeLists.txt

```diff
@@ -84,6 +84,11 @@ hermes_link_icu(libhermes)
 target_include_directories(libhermes PUBLIC ..)
 target_include_directories(libhermes PUBLIC ../../public/)
 
+target_compile_definitions(libhermes
+  PRIVATE
+  CREATE_SHARED_LIBRARY=1
```

Comment:
I am trying to figure out why this is necessary. Aren't we already setting this earlier on line 58?

Reply:
Oh, I get it. But how did it work before?

Reply:
I believe it only applies to the `hermesapi` target at the moment. 

In our app we're simply using `target_link_libraries(app libhermes)` I'm not sure if this is the expected usage? 

It appears as though `hermesapi` is the static library equivalent of `libhermes`?

Reply:
@tmikov I believe it currently works because we set `-fvisibility=default` on line 95 below, so we end up exporting everything (I don't know about the Windows case though).

Reply:
I think that flag is clang/gcc only? Adding the above triggers the `dllexport` for hermes header symbols. It's defined for the `hermesapi` target but not `libhermes`. Anyone able to confirm whether `hermesapi` is intended to basically be the static equivalent to the `libhermes` target?

Reply:
Yeah, that's how `hermesapi` is used internally. It pulls in all dependencies. IIRC, originally `libhermes` pulled in `hermesapi` as a library (which is how it should be in theory), but we ran into problems and so we just duplicated the same source files and dependencies in `libhermes`.

Reply:
Excellent thanks for the confirm. That saves me a fair amount of hassle.

Reply:
I suspect that's the root cause of the windows symbol export issues then. Perhaps a thing that was missed when linking libhermes against hermesapi was removed.